uaclient/cli/tests/test_cli_attach.py

```diff
@@ -241,8 +241,10 @@ def test_lock_file_exists(
     @mock.patch(M_PATH + "contract.UAContractClient.add_contract_machine")
     @mock.patch("uaclient.actions.status", return_value=("", 0))
     @mock.patch("uaclient.status.format_tabular")
+    @mock.patch(M_PATH + "daemon")
```

Comment:
Would it be better to mock `_post_cli_attach` like we do on the other tests ? The concern is that for some reason we add new functionality to that method, we might forget to mock it again here

Reply:
yes, but in this case another test needs to be added to cover `_post_cli_attach`, as it is being tested as part of this test.
I added this as a quick path, but I can add the extra test if that suits better

Reply:
I think that would be better tbh, since we it will be more explicit that we have a dedicated test just for that function